flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNITCase.java

```diff
@@ -101,10 +107,16 @@ public void testPerJobWithProvidedLibDirs() throws Exception {
 
 		final Configuration flinkConfig = createDefaultConfiguration(YarnConfigOptions.UserJarInclusion.DISABLED);
 		flinkConfig.set(YarnConfigOptions.PROVIDED_LIB_DIRS, Collections.singletonList(remoteLib.toString()));
-
 		runTest(() -> deployPerJob(flinkConfig, getTestingJobGraph(), false));
 	}
 
+	@Test
+	public void testPerJobWithArchive() throws Exception {
+		final Configuration flinkConfig = createDefaultConfiguration(YarnConfigOptions.UserJarInclusion.DISABLED);
+		final JobGraph archiveJobGraph = YarnTestArchiveJob.getArchiveJobGraph(tmp.newFolder(), flinkConfig);
+		runTest(() -> deployPerJob(flinkConfig, archiveJobGraph, false));
+	}
```

Comment:
The problem with the failure seems to be just that you set the last arg to `false`. I think you should set it to `true`.

Reply:
Actually the problem may be more than this. I will continue the review tomorrow.